src/controllers/update/beneficial.owner.type.controller.ts

```diff
@@ -8,14 +8,28 @@ import {
   BeneficialOwnerTypeKey,
   ManagingOfficerTypeChoice
 } from "../../model/beneficial.owner.type.model";
-import { getApplicationData } from "../../utils/application.data";
+import { BeneficialOwnerIndividualKey } from "../../model/beneficial.owner.individual.model";
+import { getApplicationData, setApplicationData } from "../../utils/application.data";
 import { ApplicationData } from "../../model";
+import { getCompanyPsc } from "../../service/persons.with.signficant.control.service";
+import { EntityNumberKey } from "../../model/data.types.model";
 
-export const get = (req: Request, res: Response, next: NextFunction) => {
+export const get = async (req: Request, res: Response, next: NextFunction) => {
   try {
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
     const appData: ApplicationData = getApplicationData(req.session);
 
+    if (appData[BeneficialOwnerIndividualKey] === undefined && appData[EntityNumberKey]) {
+      const pscData = await getCompanyPsc(req, appData[EntityNumberKey]);
```

Comment:
I get a Typescript error here if I don't have the `&& appData[EntityNumberKey]` in the if condition.

e.g.

`error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.`

Reply:
Ok I'm now getting this in concourse build:

```src/controllers/update/beneficial.owner.type.controller.ts:23:48 - error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.

23       const pscData = await getCompanyPsc(req, appData[EntityNumberKey]);```


confused :D